File: app/src/main/java/com/example/spring/ListPlanEntitlementsQueryService.java
package com.example.spring;

import com.example.spring.error.NotFoundException;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * List entitlements granted by a plan.
 */
@Service
public class ListPlanEntitlementsQueryService {

    private final EntitlementsRepository entitlementsRepository;

    public ListPlanEntitlementsQueryService(EntitlementsRepository entitlementsRepository) {
        this.entitlementsRepository = entitlementsRepository;
    }

    @Transactional(readOnly = true)
    public java.util.List<Entitlement> handle(ListPlanEntitlementsQueryRequest request) {
        // === GEN_REGION_START: service-body ===
        // spec-driven body for Query ListPlanEntitlements.
        // intent: List entitlements granted by a plan.
        // reads: EntitlementsRepository.listEntitlementsByPlan via `entitlements`.
        return this.entitlementsRepository.listEntitlementsByPlan(request.planId());
        // === GEN_REGION_END: service-body ===
    }
}

File: app-python/entitlements_app/services/list_plan_entitlements_query_service.py
"""List entitlements granted by a plan."""

from decimal import Decimal

from entitlements_app.requests.list_plan_entitlements_query_request import ListPlanEntitlementsQueryRequest
from entitlements_app.entities import Entitlement
from entitlements_app.ports import EntitlementsRepository


class ListPlanEntitlementsQueryService:
    """Service handling Query `ListPlanEntitlements`."""

    def __init__(self, entitlements_repository: EntitlementsRepository) -> None:
        self.entitlements_repository: EntitlementsRepository = entitlements_repository

    def handle(self, request: ListPlanEntitlementsQueryRequest) -> list[Entitlement]:
        # spec-driven body for Query ListPlanEntitlements.
        # intent: List entitlements granted by a plan.
        # reads: EntitlementsRepository.listEntitlementsByPlan via `entitlements`.
        return self.entitlements_repository.listEntitlementsByPlan(request.planId)
